tools/ci_build/build.py

```diff
@@ -2591,7 +2591,16 @@ def main():
     if args.build_wheel or args.gen_doc or args.use_tvm or args.enable_training:
         args.enable_pybind = True
 
-    if args.build_csharp or args.build_nuget or args.build_java or args.build_nodejs:
+    if (
+        args.build_csharp
+        or args.build_nuget
+        or args.build_java
+        or args.build_nodejs
+        or (args.enable_pybind and not args.enable_training)
+    ):
+        # If pyhon bindings are enabled, we embed the shared lib in the python package.
+        # If training is enabled, we don't embed the shared lib in the python package since training requires
+        # torch interop.
```

Comment:
This change broken our "Windows GPU Reduced Ops CI Pipeline", because previously the "--build_shared_lib" was not specified and C API related tests were not run. 
@skottmckay, how should we handle this?